posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -333,20 +334,33 @@ async def start_batch_export_run(inputs: StartBatchExportRunInputs) -> tuple[Bat
         inputs.data_interval_end,
     )
 
+    delta = dt.datetime.fromisoformat(inputs.data_interval_end) - dt.datetime.fromisoformat(inputs.data_interval_start)
     async with get_client(team_id=inputs.team_id) as client:
         if not await client.is_alive():
             raise ConnectionError("Cannot establish connection to ClickHouse")
 
-        count = await get_rows_count(
-            client=client,
-            team_id=inputs.team_id,
-            interval_start=inputs.data_interval_start,
-            interval_end=inputs.data_interval_end,
-            exclude_events=inputs.exclude_events,
-            include_events=inputs.include_events,
-        )
+        try:
+            count = await asyncio.wait_for(
+                get_rows_count(
+                    client=client,
+                    team_id=inputs.team_id,
+                    interval_start=inputs.data_interval_start,
+                    interval_end=inputs.data_interval_end,
+                    exclude_events=inputs.exclude_events,
+                    include_events=inputs.include_events,
+                ),
+                timeout=(delta / 12).total_seconds(),
```

Comment:
Selected arbitrarily, it's the same factor as our jitter.

Reply:
Happy to take suggestions, otherwise sticking with vibes.

Reply:
Makes sense. It does seem a little wasteful to spend 25 seconds (if I'm doing my math right) of CH's time every 5 minutes doing a count that will fail/cancel anyway.

I wonder if unconstrained teams should skip count attempts? Or else we could have a whole other list to skip counts for. Anyway, not critical for now it's just a thought.

Reply:
Ah, in a new comment. TLDR: I agree with you, just want to test out how this looks first, and eventually removing counts altogether.